bullet_train/docs/field-partials/dynamic-forms-dependent-fields.md

```diff
@@ -0,0 +1,123 @@
+    options: {disabled: true},
+    data: {
+      controller: "field-availability",
+      action: "dependable:updated->field-availability#toggle",
+      field_availability_expected_value: "other"
+    }
+%>
+```
+
+Note: `field-availability` here is not implemented in Bullet Train. It serves as an example.
+
+Next, we'll find a way to only serve the `:heard_from_other` field to the user if "other" is selected, this time by using server-side conditionals in a `turbo_frame`.
+
+## Dependent Fields Frame
+
+What if you'd instead want to:
+
+* Not rely on a custom Stimulus controller to control the `disabled` state of the "Other" field
+* Show/hide multiple dependent fields based on the value of the `dependable` field.
+* Update more than the field itself, but also the value of its `label`. As an example, the [`address_field`](/docs/field-partials/address-field.md) partial shows an empty "State / Province / Region" sub-field by default, and on changing the `:country_id` field to the United States, changes the whole `:region_id` to "State" as its label and with all US States as its choices.
+
+For these situations, Bullet Train has a `dependent_fields_turbo_frame` partial that's made to listen to `dependable:updated` events by default.
+
+```erb
+# update the super-select `dependable-dependents-selector-value` to "##{form.field_id(:heard_from, :dependent_fields)}" to match
+
+<%= render "shared/fields/dependent_fields_frame", 
+  id: form.field_id(:heard_from, :dependent_fields),
+  form: form,
+  dependable_fields: [:heard_from] %>
+
+  <% if form.object&.heard_from == "other" %>
+    <%# no need for a custom `id` or the `disabled` attribute %>
+    <%= render 'shared/fields/text_field', method: :heard_from_other %>
+  <% end %>
+
+  <%# include additional fields if "other" is selected %>
+<% end %>
+```
+
+This `dependent_fields_frame` serves two purposes:
+
+1. Handle the `dependable:updated` event, so that the frame can...
+2. Re-fetch the current form URL (it could be for a `#new` or a `#edit`, it works in both situations) with a GET request (not a submit) that contains the `heard_from` value as a `query_string` param. It then ensures that our `form.object.heard_from` value gets populated with the value found in the `query_string` param automatically, with **no changes needed to the resource controller**. That's all handled by the `dependent_fields_frame` partial by reading its `dependable_fields` param.
+
+With this functionality, the contents of the underlying `turbo_frame` will be populated with the updated fields.
+
+---
+
+Now let's say we want to come back to the `disabled` use case above, while using the `dependent_fields_frame` approach.
+
+We'll move the conditional on the `disabled` property. And we'll also let the `dependent_fields_frame` underlying controller handle disabling the field automatically when the `turbo_frame` awaits updates.
+
+```erb
+<%= render "shared/fields/dependent_fields_frame", 
```

Comment:
Technically not relevant to this PR, but I keep thinking I should add a default Form Builder for Bullet Train, so we can get stuff like this out of the box:

```erb
<%= form.dependent_fields_frame :heard_from do |controller| %>
  <%= form.text_field :heard_from_other, disabled: form.object&.heard_from != "other",
      data: { "#{controller}-target": "field" %>
<% end %>
```